### ISSUE TITLE
Cannot destructure property `serverRuntimeConfig` of 'undefined' or 'null'

### ISSUE THREAD

**Issue by sorleone** [zeit/next.js#8286]
Can `serverRuntimeConfig` be used when in serverless mode?
The following code:

```js
import getConfig from 'next/config';
const { serverRuntimeConfig } = getConfig();
```

returns

`Cannot destructure property serverRuntimeConfig of 'undefined' or 'null'`

For `publicRuntimeConfig` there is an error saying that it can't be used but not for that one.

**Comment by MarchWorks** [comment]
https://github.com/zeit/next.js#runtime-configuration

**Comment by MrEfrem** [comment]
> https://github.com/zeit/next.js#runtime-configuration
> 
> > > Warning: Note that this option is not available when using target: 'serverless'
> > 
> > 
> > > Warning: .................................................
> > 
> > 
> > The next/config module gives your app access to the publicRuntimeConfig and serverRuntimeConfig stored in your next.config.js.

I've the same error and I don't use `target: serverless`

**Comment by sorleone** [comment]
> > https://github.com/zeit/next.js#runtime-configuration
> > > > Warning: Note that this option is not available when using target: 'serverless'
> > > 
> > > 
> > > > Warning: .................................................
> > > 
> > > 
> > > The next/config module gives your app access to the publicRuntimeConfig and serverRuntimeConfig stored in your next.config.js.
> 
> I've the same error and I don't use `target: serverless`

Once deployed on Now or locally?

**Comment by MrEfrem** [comment]
> > > https://github.com/zeit/next.js#runtime-configuration
> > > > > Warning: Note that this option is not available when using target: 'serverless'
> > > > 
> > > > 
> > > > > Warning: .................................................
> > > > 
> > > > 
> > > > The next/config module gives your app access to the publicRuntimeConfig and serverRuntimeConfig stored in your next.config.js.
> > 
> > 
> > I've the same error and I don't use `target: serverless`
> 
> Once deployed on Now or locally?

locally. Only I already don't use this feature. I found solution with simple `env` configuration.

**Comment by rnacken** [comment]
If you are using a custom server, be sure to use `getConfig` only after initialisation of the Next app:
```
import getConfig from 'next/config';

console.log(getConfig()) // undefined - no config initialised yet
const app = next({ ... })
console.log(getConfig()) // this works
```

**Comment by Timer** [comment]
When deploying to ZEIT Now the target is forced to `serverless` -- you cannot deploy a `server` Next.js app to ZEIT Now!

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
